ci-operator/step-registry/azure/provision/service-principal/minimal-permission/azure-provision-service-principal-minimal-permission-commands.sh

```diff
@@ -41,43 +41,27 @@ function run_cmd_with_retries_save_output()
     return 0
 }
 
-function create_role_definition_json() {
-
-    local role_name=$1 permissions=$2 role_definition_file=$3
-
-    role_description="the custom role ${role_name} with minimal permissions for cluster ${CLUSTER_NAME}"
-    assignable_scopes="""
-\"/subscriptions/${AZURE_AUTH_SUBSCRIPTOIN_ID}\"
-"""
-
-    # create role definition json file
-    jq --null-input \
-       --arg role_name "${role_name}" \
-       --arg description "${role_description}" \
-       --argjson assignable_scopes "[ ${assignable_scopes} ]" \
-       --argjson permission_list "[ ${permissions} ]" '
+function run_cmd_with_retries()
 {
-  "Name": $role_name,
-  "IsCustom": true,
-  "Description": $description,
-  "assignableScopes": $assignable_scopes,
-  "Actions": $permission_list,
-  "notActions": [],
-  "dataActions": [],
-  "notDataActions": []
-}' > "${role_definition_file}"
-}
-
-function create_custom_role() {
-    local role_definition="$1"
-    local custom_role_name="$2"
-
-    # create custom role
-    cmd="az role definition create --role-definition ${role_definition}"
-    run_command "${cmd}" || return 1
+    local cmd="$1" retries="${2:-}"
+    local try=0 ret=0
+    [[ -z ${retries} ]] && max="20" || max=${retries}
+    echo "Trying ${max} times max to run '${cmd}'"
 
-    echo "Sleep 1 min to wait for custom role created"
-    sleep 60
+    res=$(eval "${cmd}") || ret=$?
```

Comment:
```suggestion
    eval "${cmd}" || ret=$?
```

Reply:
${cmd} might execute and give empty output, but return code is still 0, so here save output to check. The successful result is that output is not empty and return code is 0.